packages/core/content-releases/admin/src/components/ReleaseModal.tsx

```diff
@@ -51,24 +95,130 @@ export const ReleaseModal = ({
       >
-        {({ values, errors, handleChange }) => (
+        {({ values, errors, dirty, handleChange, setFieldValue }) => (
           <Form>
             <ModalBody>
-              <TextInput
-                label={formatMessage({
-                  id: 'content-releases.modal.form.input.label.release-name',
-                  defaultMessage: 'Name',
-                })}
-                name="name"
-                value={values.name}
-                error={errors.name}
-                onChange={handleChange}
-                required
-              />
+              <Flex direction="column" alignItems="stretch" gap={6}>
+                <TextInput
+                  label={formatMessage({
+                    id: 'content-releases.modal.form.input.label.release-name',
+                    defaultMessage: 'Name',
+                  })}
+                  name="name"
+                  value={values.name}
+                  error={errors.name}
+                  onChange={handleChange}
+                  required
+                />
+                <Checkbox
+                  name="isScheduled"
+                  value={values.isScheduled}
+                  onChange={(event) => {
+                    setFieldValue('isScheduled', event.target.checked);
+                    if (!event.target.checked) {
+                      // Clear scheduling info from a release on unchecking schedule release
+                      setFieldValue('date', null);
+                      setFieldValue('time', '');
+                    } else {
+                      setFieldValue('date', initialValues.date);
+                      setFieldValue('time', initialValues.time);
+                    }
+                  }}
+                >
+                  <Typography
+                    textColor={values.isScheduled ? 'primary600' : 'neutral800'}
+                    fontWeight={values.isScheduled ? 'semiBold' : 'regular'}
+                  >
+                    {formatMessage({
+                      id: 'modal.form.input.label.schedule-release',
+                      defaultMessage: 'Schedule release',
+                    })}
+                  </Typography>
+                </Checkbox>
+                {values.isScheduled && (
+                  <>
+                    <Flex gap={4}>
+                      <Box width="100%">
+                        <DatePicker
+                          label={formatMessage({
+                            id: 'content-releases.modal.form.input.label.date',
+                            defaultMessage: 'Date',
+                          })}
+                          name="date"
+                          onChange={(date) => {
+                            const isoFormatDate = date
+                              ? formatISO(date, { representation: 'date' })
+                              : null;
+                            setFieldValue('date', isoFormatDate);
+                          }}
```

Comment:
Can `Formik` not handle Dates as values? I'm wondering if you convert it to what the API needs before posting?

Reply:
No, it didnt work.